core/vm/contracts.go

```diff
@@ -1198,10 +1203,29 @@ func (c *l1sload) Run(state StateDB, input []byte) ([]byte, error) {
 		keys[i] = common.BytesToHash(input[20+32*i : 52+32*i])
 	}
 
-	res, err := c.l1Client.StoragesAt(context.Background(), address, keys, block)
-	if err != nil {
-		return nil, &ErrL1RPCError{err: err}
+	// if caller type is non-worker then we can retry request multiple times and return err, the tx will be reinserted in tx poll
+	// otherwise, we should retry requests forever
+	if c.callerType == CallerTypeNonWorker {
+		for {
+			res, err := c.l1Client.StoragesAt(context.Background(), address, keys, block)
+			if err == nil {
+				return res, nil
+			}
+			// wait before retrying
+			time.Sleep(100 * time.Millisecond)
+			log.Warn("L1 client request error", "err", err)
+		}
+	} else {
+		var innerErr error
+		for i := 0; i < l1ClientMaxRetries; i++ {
+			res, err := c.l1Client.StoragesAt(context.Background(), address, keys, block)
+			if err != nil {
+				innerErr = err
```

Comment:
what about printing the number of `i-th` tries, `address`, `keys`, `block` and `error messages` here? because during the middle of the for-loop, the error will be simply omitted, it's not worse recording them somehow at first.